toolkit/tools/imager/imager.go

```diff
@@ -144,6 +145,13 @@ func buildSystemConfig(systemConfig configuration.SystemConfig, disks []configur
 		return
 	}
 
+	// The old way of including mariner-release in images was reliant on some old behavior
+	// from tdnf, i.e. not detecting packages installed via rpm. Which caused an issue when
+	// tdnf was upgraded past 3.4.1 to include a package install history. Instead of pre-installing
+	// mariner-release ahead of calculating the packages + dependencies for the image, now we will
+	// simply add mariner-release to each image's package list to be installed.
```

Comment:
I'd say there's no need to reference any "old way" of doing things, since most readers won't be aware what the "old way" was from this comment. Instead, I'd focus on explaining what we're doing here given the current situation and why it is needed.